lib/error.g

```diff
@@ -59,16 +59,47 @@ end);
 
 ErrorLVars := fail;
 
-BIND_GLOBAL("WHERE", function( context, depth, outercontext)
-    local   bottom,  lastcontext,  f;
+BIND_GLOBAL("PRETTY_PRINT_VARS", function(context)
+    local vars, i;
+    vars := ContentsLVars(context);
+    if IsRecord(vars) then
+        PrintTo(ERROR_OUTPUT, "-Arguments:\n");
+        for i in [1..ABS_RAT(NumberArgumentsFunction(vars.func))] do
+            PrintTo(ERROR_OUTPUT, "   ",vars.names[i],": ");
+            if IsBound(vars.values[i]) then
+                PrintTo(ERROR_OUTPUT, vars.values[i], "\n");
+            else
+                PrintTo(ERROR_OUTPUT, "<unassigned>\n");
+            fi;
+        od;
+        PrintTo(ERROR_OUTPUT, "-Locals:\n");
+        for i in [ABS_RAT(NumberArgumentsFunction(vars.func))+1..Length(vars.names)] do
+            PrintTo(ERROR_OUTPUT, "   ",vars.names[i],": ");
+            if IsBound(vars.values[i]) then
+                PrintTo(ERROR_OUTPUT, vars.values[i], "\n");
+            else
+                PrintTo(ERROR_OUTPUT, "<unassigned>\n");
+            fi;
+        od;
+        PrintTo(ERROR_OUTPUT, "\n\n");
+    fi;
+end);
+
+BIND_GLOBAL("WHERE", function( context, depth, outercontext, showlocals)
+    local bottom, lastcontext, f, args;
     if depth <= 0 then
         return;
     fi;
     bottom := GetBottomLVars();
     lastcontext := outercontext;
     while depth > 0  and context <> bottom do
         PRINT_CURRENT_STATEMENT(ERROR_OUTPUT, context);
-        PrintTo(ERROR_OUTPUT, " called from\n");
+        if showlocals then
+            PrintTo(ERROR_OUTPUT,"\nwith local state:\n");
```

Comment:
```suggestion
            PrintTo(ERROR_OUTPUT,"\n with local state:\n");
```
To match the indentation of "called from". I wonder if indenting it even more would be even better, though (and then of course also indent the output of `PRETTY_PRINT_VARS` to match)